File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/JsonHandler.java
package closer.vlllage.com.closer.handler.helpers;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonElement;

import closer.vlllage.com.closer.pool.PoolMember;

public class JsonHandler extends PoolMember {

    private Gson gson = new GsonBuilder()
            .setDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
            .create();

    public <T> T from(String json, Class<T> clazz) {
        return gson.fromJson(json, clazz);
    }

    public String to(Object obj) {
        return gson.toJson(obj);
    }

    public JsonElement toJsonTree(Object obj) {
        return gson.toJsonTree(obj);
    }

    public <T> T from(JsonElement jsonElement, Class<T> clazz) {
        return gson.fromJson(jsonElement, clazz);
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/JsonHandler.kt
package closer.vlllage.com.closer.handler.helpers

import com.queatz.on.On
import com.google.gson.GsonBuilder
import com.google.gson.JsonElement

class JsonHandler constructor(private val on: On) {

    private val gson = GsonBuilder()
            .setDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
            .create()

    fun <T> from(json: String, clazz: Class<T>): T {
        return gson.fromJson(json, clazz)
    }

    fun to(obj: Any): String {
        return gson.toJson(obj)
    }

    fun toJsonTree(obj: Any): JsonElement {
        return gson.toJsonTree(obj)
    }

    fun <T> from(jsonElement: JsonElement, clazz: Class<T>): T {
        return gson.fromJson(jsonElement, clazz)
    }
}
